dotnetv3/Route53/Actions/Route53Wrapper.cs

```diff
@@ -0,0 +1,278 @@
+    {
+        // This example uses the same contact information for admin, registrant, and tech contacts.
+        try
+        {
+            var result = await _amazonRoute53Domains.RegisterDomainAsync(
+                new RegisterDomainRequest()
+                {
+                    AdminContact = contact,
+                    RegistrantContact = contact,
+                    TechContact = contact,
+                    DomainName = domainName,
+                    AutoRenew = autoRenew,
+                    DurationInYears = duration,
+                    PrivacyProtectAdminContact = false,
+                    PrivacyProtectRegistrantContact = false,
+                    PrivacyProtectTechContact = false
+                }
+            );
+            return result.OperationId;
+        }
+        catch (InvalidInputException)
+        {
+            _logger.LogInformation($"Unable to request registration for domain {domainName}");
+            return null;
+        }
+    }
+    // snippet-end:[Route53.dotnetv3.RegisterDomain]
+
+    // snippet-start:[Route53.dotnetv3.ViewBilling]
+
+    /// <summary>
+    /// View billing records for the account between a start and end date.
+    /// </summary>
+    /// <param name="startDate">The start date for billing results.</param>
+    /// <param name="endDate">The end date for billing results.</param>
+    /// <returns>A collection of billing records.</returns>
+    public async Task<List<BillingRecord>> ViewBilling(DateTime startDate, DateTime endDate)
+    {
+        var results = new List<BillingRecord>();
+        var paginateBilling = _amazonRoute53Domains.Paginators.ViewBilling(
+            new ViewBillingRequest()
+            {
+                Start = startDate,
+                End = endDate
+            });
+
+        // Get the entire list using the paginator.
+        await foreach (var billingRecords in paginateBilling.BillingRecords)
+        {
+            results.Add(billingRecords);
+        }
+        return results;
+    }
+    // snippet-end:[Route53.dotnetv3.ViewBilling]
+
+    // snippet-start:[Route53.dotnetv3.ListDomains]
+
+    /// <summary>
+    /// List the domains for the account.
+    /// </summary>
+    /// <returns>A collection of domain summary records.</returns>
+    public async Task<List<DomainSummary>> ListDomains()
+    {
+        var results = new List<DomainSummary>();
+        var paginateDomains = _amazonRoute53Domains.Paginators.ListDomains(
+            new ListDomainsRequest());
+
+        // Get the entire list using the paginator.
+        await foreach (var domain in paginateDomains.Domains)
+        {
+            results.Add(domain);
+        }
+        return results;
+    }
+    // snippet-end:[Route53.dotnetv3.ListDomains]
+
+    // snippet-start:[Route53.dotnetv3.ListOperations]
+
+    /// <summary>
+    /// List operations for the account, submitted since a specified date.
```

Comment:
List operations for the account that are submitted after a specified date.

Reply:
Fixed.